logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java

```diff
@@ -0,0 +1,307 @@
+
+/**
+ * A centralized log server for Alluxio
+ *
+ * Alluxio masters and workers generate logs and store the logs in local storage.
+ * {@link AlluxioLogServerProcess} allows masters and workers to "push" their logs to a
+ * centralized log server where another copy of the logs will be stored.
+ */
+public class AlluxioLogServerProcess implements LogServerProcess {
+  private static final Logger LOG = LoggerFactory.getLogger(AlluxioLogServer.class);
+  private final int mRemoteMastersLoggingPort;
+  private final int mRemoteWorkersLoggingPort;
+  private String mBaseLogsDir;
+  private Thread mMastersLogThread;
+  private Thread mWorkersLogThread;
+  private LogReceiver mMastersLogReceiver;
+  private LogReceiver mWorkersLogReceiver;
+  private int mMinNumberOfThreads;
+  private int mMaxNumberOfThreads;
+  private ExecutorService mThreadPool;
+  private volatile boolean mStopped;
+
+  private long mStopTimeoutInMillis = 60000;
+  private long mRequestTimeoutInMillis = 20000;
+  private int mBackoffSlotInMillis = 100;
+  private Random mRandom = new Random(System.currentTimeMillis());
+
+  /**
+   * Construct an {@link AlluxioLogServerProcess} instance.
+   *
+   * @param baseLogsDir base directory to store the logs pushed from remote Alluxio servers
+   */
+  public AlluxioLogServerProcess(String baseLogsDir) {
+    // TODO(yanqin) make it configurable.
+    mRemoteMastersLoggingPort = 47120;
+    mRemoteWorkersLoggingPort = 47121;
+    mMinNumberOfThreads = 8;
+    mMaxNumberOfThreads = 100;
+    mBaseLogsDir = baseLogsDir;
+    mStopped = true;
+  }
+
+  @Override
+  public void start() throws Exception {
+    SynchronousQueue<Runnable> synchronousQueue =
+        new SynchronousQueue<>();
+    mThreadPool =
+        new ThreadPoolExecutor(mMinNumberOfThreads, mMaxNumberOfThreads,
+            60, TimeUnit.SECONDS, synchronousQueue);
+    startLogServerThreads();
+  }
+
+  @Override
+  public void stop() throws Exception {
+    mStopped = true;
+    stopLogServerThreads();
+  }
+
+  @Override
+  public void waitForReady() {
+    CommonUtils.waitFor(this + " to start", new Function<Void, Boolean>() {
+      @Override
+      public Boolean apply(Void input) {
+        return mStopped == false;
+      }
+    }, WaitForOptions.defaults().setTimeoutMs(10000));
+  }
+
+  /**
+   * Create and start logging server threads.
+   */
+  private void startLogServerThreads() {
+    mMastersLogReceiver = new LogReceiver(mRemoteMastersLoggingPort, "MASTER_LOGGER");
+    mWorkersLogReceiver = new LogReceiver(mRemoteWorkersLoggingPort, "WORKER_LOGGER");
+    mMastersLogThread = new Thread(mMastersLogReceiver);
+    mWorkersLogThread = new Thread(mWorkersLogReceiver);
+    mStopped = false;
+    mMastersLogThread.start();
+    mWorkersLogThread.start();
+    while (!Thread.interrupted()) {
+      CommonUtils.sleepMs(LOG, 1000);
+    }
+  }
+
+  private void stopLogServerThreads() {
+    mMastersLogReceiver.close();
+    mWorkersLogReceiver.close();
```

Comment:
shouldn't `mStopped` be set in this method, to mirror `startLogServerThreads()`?